crates/optimism/payload/src/builder.rs

```diff
@@ -496,7 +501,11 @@ where
         logs_bloom,
         timestamp: attributes.payload_attributes.timestamp,
         mix_hash: attributes.payload_attributes.prev_randao,
-        nonce: BEACON_NONCE.into(),
+        nonce: get_nonce(
+            is_holocene,
+            &attributes,
+            chain_spec.base_fee_params_at_timestamp(attributes.payload_attributes.timestamp),
```

Comment:
My idea is that it will use `base_fee_params_at_timestamp` as the default values since it's not modified. For each block it will only fetch from `nonce` for the params. Lmk if this makes sense at all

Reply:
I believe this is correct. Would appreciate if @clabby could verify